lib-rt/CPy.h

```diff
@@ -321,13 +317,9 @@ static CPyTagged CPyTagged_StealFromObject(PyObject *object) {
 
 static CPyTagged CPyTagged_BorrowFromObject(PyObject *object) {
     int overflow;
-    // TODO: This may call __int__ and raise exceptions.
-    PY_LONG_LONG value = PyLong_AsLongLongAndOverflow(object, &overflow);
-    // We use a Python object if the value shifted left by 1 is too
-    // large for long long.  The latter check is micro-optimized where
-    // the common case where long long is small enough.
-    if (overflow != 0 || (((unsigned long long)value >= (1LL << 62)) &&
-                          (value >= 0 || value < -(1LL << 62)))) {
+    // The overflow check knows about CPyTagged's width
+    PY_LONG_LONG value = CPyLong_AsLongLongAndOverflow(object, &overflow);
+    if (overflow != 0) {
```

Comment:
I wonder if merging this function and `CPyLong_AsLongLongAndOverflow` would bring any benefits, or if the C compilers are able to optimize all overhead away. Since the function is called from two places, it would only be helpful if the savings are significant in a micro-benchmark at least. 

Alternatively, we could only move the fast path here, and share the rest. Maybe something like this:

```c
PyLongObject *v = (PyLongObject *)object;
if (Py_SIZE(v) == 1) {
    return v->ob_digit[0] << 1;
} else {
    return CPy_whatever(object); // Handle other cases
}
```

Again, this is really only useful if there is a measurable performance improvement in a micronbechmark.

Reply:
Couldn't get this to help